hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -33,7 +33,7 @@ def ensure_packages(packages, state = 'installed')
 
   def apply_puppet_code(code)
     bin_path = Kafo::PuppetCommand.search_puppet_path('puppet')
-    Open3.capture3("echo \"#{code}\" | #{bin_path} apply --detailed-exitcodes")
+    Open3.capture3(PuppetCommand.format_command("echo \"#{code}\" | #{bin_path} apply --detailed-exitcodes"))
```

Comment:
I've been wondering if we should provide this entire method as a higher level abstraction from Kafo. Thoughts?

Reply:
I thought about it. Right now we, the installer, use this in two places: kafo and kafo_parsers. Kafo requires kafo_parsers so I contemplated putting it all in kafo_parsers (that just felt weird). This only needs to live for the lifespan of EL7 which has only a few Foreman releases left. And really, I see packaging the installer for SCL as a way to get the installer to drop Ruby 2.0 conventions and align El7 and EL8 Ruby versions going forward. So I am inclined towards the least amount of overhead to capture this functionality. Which I why I arrived at putting it in kafo_parsers where its needed and Kafo and then making sure the installer uses Kafo (since it already relies heavily on it).